src/realm/table.hpp

```diff
@@ -234,6 +234,8 @@ class Table {
 
     /// Object handling.
 
+    enum class UpdateMode { never, changed, all };
```

Comment:
what is the purpose of this? what use cases do we expect to cover with it?

Reply:
To make something similar to [Object::create](https://github.com/realm/realm-core/blob/472d325a2e844fcb48c400db74e6b9646f92ca93/src/realm/object-store/object_accessor.hpp#L255-L257)  available in the C API.